File: templates/Customer-Support-Template/CustomerSupportTemplate/Dialogs/Cancel/CancelResponses.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using CustomerSupportTemplate.Dialogs.Cancel.Resources;
using Microsoft.Bot.Builder.TemplateManager;

namespace CustomerSupportTemplate
{
    public class CancelResponses : TemplateManager
    {
        private static LanguageTemplateDictionary _responseTemplates = new LanguageTemplateDictionary
        {
            ["default"] = new TemplateIdMap
            {
                { ResponseIds.ConfirmCancelPrompt, (context, data) => CancelStrings.ConfirmCancelPrompt },
                { ResponseIds.CancelConfirmedMessage, (context, data) => CancelStrings.CancelConfirmedMessage },
                { ResponseIds.CancelDeniedMessage, (context, data) => CancelStrings.CancelDeniedMessage },
            },
            ["en"] = new TemplateIdMap { },
            ["fr"] = new TemplateIdMap { },
        };

        public CancelResponses()
        {
            Register(new DictionaryRenderer(_responseTemplates));
        }

        public class ResponseIds
        {
            public const string ConfirmCancelPrompt = "Cancel.ConfirmCancelPrompt";
            public const string CancelConfirmedMessage = "Cancel.CancelConfirmed";
            public const string CancelDeniedMessage = "Cancel.CancelDenied";
        }
    }
}

File: templates/Enterprise-Template/src/typescript/generator-botbuilder-enterprise/generators/app/templates/enterprise-bot/src/dialogs/cancel/cancelResponses.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License

import { TurnContext } from "botbuilder";
import { DictionaryRenderer, LanguageTemplateDictionary, TemplateFunction } from "../templateManager/dictionaryRenderer";
import { TemplateManager } from "../templateManager/templateManager";
import * as i18n from "i18n";

export class CancelResponses extends TemplateManager {

    // Fields
    public static ResponseIds = {
        CancelPrompt:  "cancelPrompt",
        CancelConfirmedMessage: "cancelConfirmed",
        CancelDeniedMessage: "cancelDenied",
    }

    private static readonly _responseTemplates: LanguageTemplateDictionary = new Map([
        ["default", new Map([
            [CancelResponses.ResponseIds.CancelPrompt, CancelResponses.fromResources("cancel.prompt")],
            [CancelResponses.ResponseIds.CancelConfirmedMessage, CancelResponses.fromResources("cancel.confirmed")],
            [CancelResponses.ResponseIds.CancelDeniedMessage, CancelResponses.fromResources("cancel.denied")],
        ])]
    ]);

    private static fromResources(name: string): TemplateFunction {
        return (context: TurnContext, data: any) => Promise.resolve(i18n.__(name));
    }

    constructor() {
        super();
        this.register(new DictionaryRenderer(CancelResponses._responseTemplates));
    }
}
